src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/XdocGenerator.java

```diff
@@ -68,6 +70,11 @@ public static void generateXdocContent() throws Exception {
             finally {
                 sink.close();
             }
+            final File outputFile = new File(pathToFile.replace(".template", ""));
+            final StandardCopyOption exp = StandardCopyOption.REPLACE_EXISTING;
+            Files.copy(tempFile.toPath(), outputFile.toPath(), exp);
+            tempFile.delete();
```

Comment:
@romani this is how we delete temp file.

Reply:
If exception happens, this file will not be deleted by our code, I hope jvm will do it, in worse case it will stay in drive until OS remove it.

Reply:
VM won't delete a file if we are in full control of it.
https://stackoverflow.com/questions/16691437/when-are-java-temporary-files-deleted

Reply:
Makes sense todo:
```
File temp = File.createTempFile("abc", ".tmp"); 
temp.deleteOnExit()
```

Reply:
>If exception happens, this file will not be deleted by our code

This is why I mentioned to do this in the finally block.

Reply:
@rnveach @romani Do I need to add the last statements in the finally block ?

Reply:
I like `temp.deleteOnExit()`

Reply:
Done @romani